autoPyTorch/datasets/base_dataset.py

```diff
@@ -312,22 +324,30 @@ def create_holdout_val_split(
             self.random_state, val_share, self._get_indices(), **kwargs)
         return train, val
 
-    def get_dataset_for_training(self, split_id: int) -> Tuple[Dataset, Dataset]:
+    def get_dataset_for_training(self, split_id: int, train: bool) -> Dataset:
         """
         The above split methods employ the Subset to internally subsample the whole dataset.
 
         During training, we need access to one of those splits. This is a handy function
         to provide training data to fit a pipeline
 
         Args:
-            split (int): The desired subset of the dataset to split and use
+            split_id (int): which split id to get from the splits
+            train (bool): whether the dataset is required for training or evaluating.
 
         Returns:
             Dataset: the reduced dataset to be used for testing
         """
         # Subset creates a dataset. Splits is a (train_indices, test_indices) tuple
-        return (TransformSubset(self, self.splits[split_id][0], train=True),
-                TransformSubset(self, self.splits[split_id][1], train=False))
+        if split_id >= len(self.splits):  # old version: split_id > len(self.splits)
+            raise IndexError("split_id out of range, got split_id={}"
+                             " (>= num_splits={})".format(split_id, len(self.splits)))
+        subset = int(not train)
+        indices = self.splits[split_id][subset]
+        if indices is None:
```

Comment:
Does that happen?
I mean wouldn't we get an error before coming here if it is raised?

Reply:
This happens when we are using `NoResamplingStrategy`. It does not split the training data, it returns None for `val_indices`.

Reply:
Thanks, I got the algorithm.